atomix/cluster/src/main/java/io/atomix/raft/protocol/VersionedAppendRequest.java

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright 2015-present Open Networking Foundation
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.protocol;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.atomix.cluster.MemberId;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Append entries request that represent new versions (version > 1)
+ *
+ * <p>Append entries requests are at the core of the replication protocol. Leaders send append
+ * requests to followers to replicate and commit log entries, and followers sent append requests to
+ * passive members to replicate committed log entries.
+ */
+public class VersionedAppendRequest extends AbstractRaftRequest {
+
+  private static final int CURRENT_VERSION = 2;
+
+  private final int version;
+  private final long term;
+  private final String leader;
+  private final long prevLogIndex;
+  private final long prevLogTerm;
+  private final List<PersistedRaftRecord> entries;
+  private final long commitIndex;
+
+  public VersionedAppendRequest(
+      final int version,
+      final long term,
+      final String leader,
+      final long prevLogIndex,
+      final long prevLogTerm,
+      final List<PersistedRaftRecord> entries,
```

Comment:
:question: This will become `PersistedJournalRecord` right?

Reply:
Yes.